### ISSUE TITLE
stylus-loader >= 4 broken: Invalid options object

### ISSUE THREAD

**Issue by tiltec** [quasarframework/quasar#10448]
**Describe the bug**

I followed the Quasar v1 -> v2 upgrade guide and decided to keep stylus-loader in my project for now. After running `yarn add --dev stylus stylus-loader` and starting the dev server, I get this error

```
ValidationError: Invalid options object. Stylus Loader has been initialized using an options object that does not match the API schema.
   - options has an unknown property 'preferPathResolver'. These properties are valid:
     object { implementation?, stylusOptions?, sourceMap?, webpackImporter?, additionalData? }
```

It seems to boil down to this line in `@quasar/app`:

https://github.com/quasarframework/quasar/blob/d046f4cf235c32a2c1d2ab093c42b1fc5ae2355e/app/lib/webpack/inject.style-rules.js#L180

When I remove it, `quasar dev` works fine with `stylus-loader@6`.

`preferPathResolver` seems to have been removed starting in `4.0.0`, although unfortunately this isn't mentioned in their [changelog](https://github.com/webpack-contrib/stylus-loader/blob/master/CHANGELOG.md).

This commit in `stylus-loader` removes the section from `README.md`:

https://github.com/webpack-contrib/stylus-loader/commit/dcf4daac3c12a9a10af3b58c841e4cae3cfb4cfd#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L183-L185

**Codepen/jsFiddle/Codesandbox (required)**

https://codesandbox.io/s/intelligent-glade-ftvgl

**To Reproduce**
run `quasar dev` and check the resulting error

**Expected behavior**
process `stylus` styles

**Screenshots**

![image](https://user-images.githubusercontent.com/4410802/130363812-9bf7f7a4-62dd-4154-bec4-98869d2baba2.png)

**Platform (please complete the following information):**
Quasar Version: 2.0.4
@quasar/app Version: 3.1.0
Quasar mode:
  - [x] SPA
  - [ ] SSR
  - [ ] PWA
  - [ ] Electron
  - [ ] Cordova
  - [ ] Capacitor
  - [ ] BEX

Tested on:
  - [x] SPA
  - [ ] SSR
  - [ ] PWA
  - [ ] Electron
  - [ ] Cordova
  - [ ] Capacitor
  - [ ] BEX

OS: Linux
Node: 14
NPM: 7.20
Yarn: 1.22

**Comment by sschneider-ihre-pvs** [comment]
https://github.com/webpack-contrib/stylus-loader/commit/da566a6#r56565818

**Comment by rstoenescu** [comment]
Fix will be available in "@quasar/app" v3.1.3
